torch/serialization.py

```diff
@@ -475,6 +477,7 @@ def persistent_id(obj):
                     location,
                     obj.size())
         return None
+    persistent_id.id_map = {}
```

Comment:
Not sure why you bind this to function. Could you explain? IMO it makes it less readable. If there is no "good" reason, I would simply add it as local variable below `serialized_storages` (L458).

Reply:
The reason was that I thought of `id_map` as a C-style static variable bound to a function (e.g. see [here](https://www.geeksforgeeks.org/static-variables-in-c/)).

Logically, `id_map` is only used by the inner function `persistent_id()` and nowhere in `_save()` we really need to access or even care about the existence of `id_map`. On the contrary, `serialized_storages` is modified by `persistent_id()` but it is also used by `_save()`, so I think that is a slightly different case.

Anyway, in terms of output the two approaches give the same result so if you prefer to move the variable to the outer scope I will do it, no problem!

Reply:
Fixed in 1e4cfc7386.